modules/testserver/src/main/java/com/smartbear/readyapi4j/testserver/execution/TestServerRecipeExecutor.java

```diff
@@ -58,17 +57,17 @@ public TestServerExecution executeRecipe(TestRecipe recipe) {
             recipeFilter.filterRecipe(recipe);
         }
 
-        TestServerExecution execution = doExecuteTestCase(recipe.getTestCase(), recipe.getExtractorData(), false);
+        TestServerExecution execution = doExecuteTestCase(recipe, recipe.getExtractorData(), false);
         notifyExecutionFinished(execution);
         return execution;
     }
 
-    private TestServerExecution doExecuteTestCase(TestCase testCase, ExtractorData optionalExtractorData, boolean async)  {
+    private TestServerExecution doExecuteTestCase(TestRecipe testRecipe, ExtractorData optionalExtractorData, boolean async)  {
         try {
             Optional<ExtractorData> extractorDataOptional = Optional.ofNullable(optionalExtractorData);
             extractorDataOptional.ifPresent(extractorData -> extractorDataList.add(extractorData));
-            TestServerExecution execution = testServerClient.postTestRecipe(testCase, async);
-            cancelExecutionAndThrowExceptionIfPendingDueToMissingClientCertificate(execution.getCurrentReport(), testCase);
+            TestServerExecution execution = testServerClient.postTestRecipe(testRecipe, async);
+            cancelExecutionAndThrowExceptionIfPendingDueToMissingClientCertificate(execution.getCurrentReport(), testRecipe.getTestCase());
```

Comment:
That method name is a mouthful ... but maybe it's a good one. :-) And it's not new anyway.